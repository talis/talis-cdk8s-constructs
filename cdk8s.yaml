language: typescript
imports:
  - k8s@1.30.0
