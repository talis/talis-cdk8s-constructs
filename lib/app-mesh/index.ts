export * from "./ingress-gateway-nlb";
export * from "./app-service";
