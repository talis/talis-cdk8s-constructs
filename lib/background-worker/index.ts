export * from "./background-worker-props";
export * from "./background-worker";
