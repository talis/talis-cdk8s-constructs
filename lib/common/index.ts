export * from "./container-props";
export * from "./env-util";
export * from "./name-util";
export * from "./statefulset-util";
export * from "./value-util";
export * from "./workload-props";
