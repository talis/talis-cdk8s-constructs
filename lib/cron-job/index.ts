export * from "./cron-job-props";
export * from "./cron-job";
