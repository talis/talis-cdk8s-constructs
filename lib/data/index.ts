export * from "./config-map";
export * from "./secret";
export * from "./factory";
