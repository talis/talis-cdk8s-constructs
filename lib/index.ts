export * from "./cron-job";
export * from "./background-worker";
export * from "./data";
export * from "./job";
export * from "./redis";
export * from "./talis-chart";
export * from "./web-service";
