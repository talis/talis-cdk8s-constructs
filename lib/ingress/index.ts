export * from "./ingress-props";
export * from "./ingress";
