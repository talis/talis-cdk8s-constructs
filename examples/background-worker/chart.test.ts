import { BackgroundWorkerChart } from "./chart";
import { Testing } from "cdk8s";
import { TalisShortRegion, TalisDeploymentEnvironment } from "../../lib";

describe("BackgroundWorker example", () => {
  const PROCESS_ENV = process.env;

  beforeEach(() => {
    jest.resetModules();
    process.env = { ...PROCESS_ENV };
    process.env.DOCKER_USERNAME = "someuser";
    process.env.DOCKER_PASSWORD = "secret123";
  });

  afterEach(() => {
    process.env = PROCESS_ENV;
  });

  test("Snapshot", () => {
    const app = Testing.app();
    const chart = new BackgroundWorkerChart(app, {
      environment: TalisDeploymentEnvironment.DEVELOPMENT,
      region: TalisShortRegion.LOCAL,
      watermark: "test",
    });
    const results = Testing.synth(chart);
    expect(results).toMatchSnapshot();
  });

  test("Change release version", () => {
    process.env.RELEASE = "v2.2";
    const app = Testing.app();
    const chart = new BackgroundWorkerChart(app, {
      environment: TalisDeploymentEnvironment.DEVELOPMENT,
      region: TalisShortRegion.LOCAL,
      watermark: "test",
    });
    const results = Testing.synth(chart);
    expect(results).toMatchSnapshot();
  });
});
