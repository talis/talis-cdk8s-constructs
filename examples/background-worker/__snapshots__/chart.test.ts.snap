// Vitest Snapshot v1

exports[`BackgroundWorker example > Change release version 1`] = `
[
  {
    "apiVersion": "v1",
    "kind": "Namespace",
    "metadata": {
      "labels": {
        "app": "worker",
        "environment": "development",
        "managed-by": "cdk8s",
        "region": "local",
        "service": "worker-development-local",
      },
      "name": "worker-test",
      "namespace": "worker-test",
    },
  },
  {
    "apiVersion": "v1",
    "data": {
      ".dockerconfigjson": "eyJhdXRocyI6eyJodHRwczovL2luZGV4LmRvY2tlci5pby92MS8iOnsiYXV0aCI6ImMyOXRaWFZ6WlhJNmMyVmpjbVYwTVRJeiJ9fX0=",
    },
    "kind": "Secret",
    "metadata": {
      "labels": {
        "app": "worker",
        "environment": "development",
        "managed-by": "cdk8s",
        "region": "local",
        "service": "worker-development-local",
      },
      "name": "docker-hub-cred",
      "namespace": "worker-test",
    },
    "type": "kubernetes.io/dockerconfigjson",
  },
  {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": {
      "annotations": {
        "talis.io/chat": "https://talis.slack.com/archives/C04P9DPCX",
        "talis.io/description": "Frontend to the Resque job queue system",
        "talis.io/eks-dashboard": "None",
        "talis.io/graphs": "None",
        "talis.io/incidents": "None",
        "talis.io/issues": "https://github.com/talis/platform/issues",
        "talis.io/logs": "None",
        "talis.io/repository": "https://github.com/talis/resque-web-container",
        "talis.io/runbook": "None",
        "talis.io/uptime": "None",
        "talis.io/url": "https://resque.example.com",
      },
      "labels": {
        "app": "resque",
        "environment": "development",
        "instance": "resque",
        "managed-by": "cdk8s",
        "region": "local",
        "release": "stable",
        "role": "server",
        "service": "worker-development-local",
      },
      "name": "resque-service",
      "namespace": "worker-test",
    },
    "spec": {
      "ports": [
        {
          "port": 3000,
          "protocol": "TCP",
          "targetPort": 3000,
        },
      ],
      "selector": {
        "app": "resque",
        "instance": "resque",
        "role": "server",
      },
      "type": "NodePort",
    },
  },
  {
    "apiVersion": "networking.k8s.io/v1",
    "kind": "Ingress",
    "metadata": {
      "annotations": {
        "alb.ingress.kubernetes.io/listen-ports": "[{\\"HTTP\\":80},{\\"HTTPS\\":443}]",
        "alb.ingress.kubernetes.io/load-balancer-attributes": "idle_timeout.timeout_seconds=60",
        "alb.ingress.kubernetes.io/load-balancer-name": "worker-test-resque-develop",
        "alb.ingress.kubernetes.io/ssl-policy": "ELBSecurityPolicy-TLS-1-2-2017-01",
        "alb.ingress.kubernetes.io/ssl-redirect": "443",
        "alb.ingress.kubernetes.io/success-codes": "200,303",
        "alb.ingress.kubernetes.io/tags": "service=worker-development-local,instance=resque,environment=development",
        "alb.ingress.kubernetes.io/target-type": "instance",
      },
      "labels": {
        "app": "resque",
        "environment": "development",
        "instance": "resque",
        "managed-by": "cdk8s",
        "region": "local",
        "release": "stable",
        "role": "server",
        "service": "worker-development-local",
      },
      "name": "resque-ingress",
      "namespace": "worker-test",
    },
    "spec": {
      "defaultBackend": {
        "service": {
          "name": "resque-service",
          "port": {
            "number": 3000,
          },
        },
      },
      "ingressClassName": "aws-load-balancer-internal",
      "tls": [
        {
          "hosts": [
            "*.example.com",
          ],
        },
      ],
    },
  },
  {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": {
      "labels": {
        "app": "resque",
        "environment": "development",
        "instance": "resque",
        "managed-by": "cdk8s",
        "region": "local",
        "release": "stable",
        "role": "server",
        "service": "worker-development-local",
      },
      "name": "resque",
      "namespace": "worker-test",
    },
    "spec": {
      "replicas": 1,
      "revisionHistoryLimit": 1,
      "selector": {
        "matchLabels": {
          "app": "resque",
          "instance": "resque",
          "role": "server",
        },
      },
      "template": {
        "metadata": {
          "labels": {
            "app": "resque",
            "environment": "development",
            "instance": "resque",
            "managed-by": "cdk8s",
            "region": "local",
            "release": "stable",
            "role": "server",
            "service": "worker-development-local",
          },
        },
        "spec": {
          "affinity": {
            "podAntiAffinity": {
              "preferredDuringSchedulingIgnoredDuringExecution": [
                {
                  "podAffinityTerm": {
                    "labelSelector": {
                      "matchLabels": {
                        "app": "resque",
                        "instance": "resque",
                        "role": "server",
                      },
                    },
                    "topologyKey": "topology.kubernetes.io/zone",
                  },
                  "weight": 100,
                },
              ],
            },
          },
          "automountServiceAccountToken": false,
          "containers": [
            {
              "env": [
                {
                  "name": "RAILS_RESQUE_REDIS",
                  "value": "redis.cache.svc.cluster.local:6379:1",
                },
              ],
              "image": "talis/resque-web:stable",
              "imagePullPolicy": "IfNotPresent",
              "name": "worker",
              "ports": [
                {
                  "containerPort": 3000,
                  "protocol": "TCP",
                },
              ],
              "resources": {
                "limits": {
                  "cpu": "100m",
                  "memory": "200Mi",
                },
                "requests": {
                  "cpu": "50m",
                  "memory": "100Mi",
                },
              },
            },
          ],
          "imagePullSecrets": [
            {
              "name": "docker-hub-cred",
            },
          ],
          "priorityClassName": "web",
        },
      },
    },
  },
  {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": {
      "labels": {
        "app": "worker",
        "environment": "development",
        "instance": "php-worker-example",
        "managed-by": "cdk8s",
        "region": "local",
        "release": "v2.2",
        "role": "worker",
        "service": "worker-development-local",
      },
      "name": "php-worker-example",
      "namespace": "worker-test",
    },
    "spec": {
      "revisionHistoryLimit": 1,
      "selector": {
        "matchLabels": {
          "app": "worker",
          "instance": "php-worker-example",
          "role": "worker",
        },
      },
      "template": {
        "metadata": {
          "labels": {
            "app": "worker",
            "environment": "development",
            "instance": "php-worker-example",
            "managed-by": "cdk8s",
            "region": "local",
            "release": "v2.2",
            "role": "worker",
            "service": "worker-development-local",
          },
        },
        "spec": {
          "affinity": {
            "podAntiAffinity": {
              "preferredDuringSchedulingIgnoredDuringExecution": [
                {
                  "podAffinityTerm": {
                    "labelSelector": {
                      "matchLabels": {
                        "app": "worker",
                        "instance": "php-worker-example",
                        "role": "worker",
                      },
                    },
                    "topologyKey": "topology.kubernetes.io/zone",
                  },
                  "weight": 100,
                },
              ],
            },
          },
          "automountServiceAccountToken": false,
          "containers": [
            {
              "command": [
                "php",
                "vendor/resque/php-resque/bin/resque",
              ],
              "env": [
                {
                  "name": "WORKER_NUM",
                  "valueFrom": {
                    "fieldRef": {
                      "fieldPath": "metadata.name",
                    },
                  },
                },
                {
                  "name": "REDIS_BACKEND",
                  "value": "redis.cache.svc.cluster.local:6379:1",
                },
                {
                  "name": "QUEUE",
                  "value": "my-app:example:jobs",
                },
              ],
              "image": "docker.io/organization/my-app:worker-v2.2",
              "imagePullPolicy": "IfNotPresent",
              "lifecycle": {
                "preStop": {
                  "exec": {
                    "command": [
                      "/bin/sh",
                      "-c",
                      "kill -QUIT 1 && while kill -0 1; do sleep 1; done",
                    ],
                  },
                },
              },
              "name": "worker",
              "resources": {
                "requests": {
                  "cpu": "50m",
                  "memory": "100Mi",
                },
              },
            },
          ],
          "imagePullSecrets": [
            {
              "name": "docker-hub-cred",
            },
          ],
          "terminationGracePeriodSeconds": 300,
        },
      },
    },
  },
]
`;

exports[`BackgroundWorker example > Snapshot 1`] = `
[
  {
    "apiVersion": "v1",
    "kind": "Namespace",
    "metadata": {
      "labels": {
        "app": "worker",
        "environment": "development",
        "managed-by": "cdk8s",
        "region": "local",
        "service": "worker-development-local",
      },
      "name": "worker-test",
      "namespace": "worker-test",
    },
  },
  {
    "apiVersion": "v1",
    "data": {
      ".dockerconfigjson": "eyJhdXRocyI6eyJodHRwczovL2luZGV4LmRvY2tlci5pby92MS8iOnsiYXV0aCI6ImMyOXRaWFZ6WlhJNmMyVmpjbVYwTVRJeiJ9fX0=",
    },
    "kind": "Secret",
    "metadata": {
      "labels": {
        "app": "worker",
        "environment": "development",
        "managed-by": "cdk8s",
        "region": "local",
        "service": "worker-development-local",
      },
      "name": "docker-hub-cred",
      "namespace": "worker-test",
    },
    "type": "kubernetes.io/dockerconfigjson",
  },
  {
    "apiVersion": "v1",
    "kind": "Service",
    "metadata": {
      "annotations": {
        "talis.io/chat": "https://talis.slack.com/archives/C04P9DPCX",
        "talis.io/description": "Frontend to the Resque job queue system",
        "talis.io/eks-dashboard": "None",
        "talis.io/graphs": "None",
        "talis.io/incidents": "None",
        "talis.io/issues": "https://github.com/talis/platform/issues",
        "talis.io/logs": "None",
        "talis.io/repository": "https://github.com/talis/resque-web-container",
        "talis.io/runbook": "None",
        "talis.io/uptime": "None",
        "talis.io/url": "https://resque.example.com",
      },
      "labels": {
        "app": "resque",
        "environment": "development",
        "instance": "resque",
        "managed-by": "cdk8s",
        "region": "local",
        "release": "stable",
        "role": "server",
        "service": "worker-development-local",
      },
      "name": "resque-service",
      "namespace": "worker-test",
    },
    "spec": {
      "ports": [
        {
          "port": 3000,
          "protocol": "TCP",
          "targetPort": 3000,
        },
      ],
      "selector": {
        "app": "resque",
        "instance": "resque",
        "role": "server",
      },
      "type": "NodePort",
    },
  },
  {
    "apiVersion": "networking.k8s.io/v1",
    "kind": "Ingress",
    "metadata": {
      "annotations": {
        "alb.ingress.kubernetes.io/listen-ports": "[{\\"HTTP\\":80},{\\"HTTPS\\":443}]",
        "alb.ingress.kubernetes.io/load-balancer-attributes": "idle_timeout.timeout_seconds=60",
        "alb.ingress.kubernetes.io/load-balancer-name": "worker-test-resque-develop",
        "alb.ingress.kubernetes.io/ssl-policy": "ELBSecurityPolicy-TLS-1-2-2017-01",
        "alb.ingress.kubernetes.io/ssl-redirect": "443",
        "alb.ingress.kubernetes.io/success-codes": "200,303",
        "alb.ingress.kubernetes.io/tags": "service=worker-development-local,instance=resque,environment=development",
        "alb.ingress.kubernetes.io/target-type": "instance",
      },
      "labels": {
        "app": "resque",
        "environment": "development",
        "instance": "resque",
        "managed-by": "cdk8s",
        "region": "local",
        "release": "stable",
        "role": "server",
        "service": "worker-development-local",
      },
      "name": "resque-ingress",
      "namespace": "worker-test",
    },
    "spec": {
      "defaultBackend": {
        "service": {
          "name": "resque-service",
          "port": {
            "number": 3000,
          },
        },
      },
      "ingressClassName": "aws-load-balancer-internal",
      "tls": [
        {
          "hosts": [
            "*.example.com",
          ],
        },
      ],
    },
  },
  {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": {
      "labels": {
        "app": "resque",
        "environment": "development",
        "instance": "resque",
        "managed-by": "cdk8s",
        "region": "local",
        "release": "stable",
        "role": "server",
        "service": "worker-development-local",
      },
      "name": "resque",
      "namespace": "worker-test",
    },
    "spec": {
      "replicas": 1,
      "revisionHistoryLimit": 1,
      "selector": {
        "matchLabels": {
          "app": "resque",
          "instance": "resque",
          "role": "server",
        },
      },
      "template": {
        "metadata": {
          "labels": {
            "app": "resque",
            "environment": "development",
            "instance": "resque",
            "managed-by": "cdk8s",
            "region": "local",
            "release": "stable",
            "role": "server",
            "service": "worker-development-local",
          },
        },
        "spec": {
          "affinity": {
            "podAntiAffinity": {
              "preferredDuringSchedulingIgnoredDuringExecution": [
                {
                  "podAffinityTerm": {
                    "labelSelector": {
                      "matchLabels": {
                        "app": "resque",
                        "instance": "resque",
                        "role": "server",
                      },
                    },
                    "topologyKey": "topology.kubernetes.io/zone",
                  },
                  "weight": 100,
                },
              ],
            },
          },
          "automountServiceAccountToken": false,
          "containers": [
            {
              "env": [
                {
                  "name": "RAILS_RESQUE_REDIS",
                  "value": "redis.cache.svc.cluster.local:6379:1",
                },
              ],
              "image": "talis/resque-web:stable",
              "imagePullPolicy": "IfNotPresent",
              "name": "worker",
              "ports": [
                {
                  "containerPort": 3000,
                  "protocol": "TCP",
                },
              ],
              "resources": {
                "limits": {
                  "cpu": "100m",
                  "memory": "200Mi",
                },
                "requests": {
                  "cpu": "50m",
                  "memory": "100Mi",
                },
              },
            },
          ],
          "imagePullSecrets": [
            {
              "name": "docker-hub-cred",
            },
          ],
          "priorityClassName": "web",
        },
      },
    },
  },
  {
    "apiVersion": "apps/v1",
    "kind": "Deployment",
    "metadata": {
      "labels": {
        "app": "worker",
        "environment": "development",
        "instance": "php-worker-example",
        "managed-by": "cdk8s",
        "region": "local",
        "release": "v1.0",
        "role": "worker",
        "service": "worker-development-local",
      },
      "name": "php-worker-example",
      "namespace": "worker-test",
    },
    "spec": {
      "revisionHistoryLimit": 1,
      "selector": {
        "matchLabels": {
          "app": "worker",
          "instance": "php-worker-example",
          "role": "worker",
        },
      },
      "template": {
        "metadata": {
          "labels": {
            "app": "worker",
            "environment": "development",
            "instance": "php-worker-example",
            "managed-by": "cdk8s",
            "region": "local",
            "release": "v1.0",
            "role": "worker",
            "service": "worker-development-local",
          },
        },
        "spec": {
          "affinity": {
            "podAntiAffinity": {
              "preferredDuringSchedulingIgnoredDuringExecution": [
                {
                  "podAffinityTerm": {
                    "labelSelector": {
                      "matchLabels": {
                        "app": "worker",
                        "instance": "php-worker-example",
                        "role": "worker",
                      },
                    },
                    "topologyKey": "topology.kubernetes.io/zone",
                  },
                  "weight": 100,
                },
              ],
            },
          },
          "automountServiceAccountToken": false,
          "containers": [
            {
              "command": [
                "php",
                "vendor/resque/php-resque/bin/resque",
              ],
              "env": [
                {
                  "name": "WORKER_NUM",
                  "valueFrom": {
                    "fieldRef": {
                      "fieldPath": "metadata.name",
                    },
                  },
                },
                {
                  "name": "REDIS_BACKEND",
                  "value": "redis.cache.svc.cluster.local:6379:1",
                },
                {
                  "name": "QUEUE",
                  "value": "my-app:example:jobs",
                },
              ],
              "image": "docker.io/organization/my-app:worker-v1.0",
              "imagePullPolicy": "IfNotPresent",
              "lifecycle": {
                "preStop": {
                  "exec": {
                    "command": [
                      "/bin/sh",
                      "-c",
                      "kill -QUIT 1 && while kill -0 1; do sleep 1; done",
                    ],
                  },
                },
              },
              "name": "worker",
              "resources": {
                "requests": {
                  "cpu": "50m",
                  "memory": "100Mi",
                },
              },
            },
          ],
          "imagePullSecrets": [
            {
              "name": "docker-hub-cred",
            },
          ],
          "terminationGracePeriodSeconds": 300,
        },
      },
    },
  },
]
`;
